Classes/Emogrifier.php

```diff
@@ -373,11 +373,21 @@ protected function process(\DOMDocument $xmlDocument)
         $excludedNodes = $this->getNodesToExclude($xPath);
         $cssRules = $this->parseCssRules($cssParts['css']);
         foreach ($cssRules as $cssRule) {
-            // query the body for the xpath selector
-            $nodesMatchingCssSelectors = $xPath->query($this->translateCssToXpath($cssRule['selector']));
-            // ignore invalid selectors
-            if ($nodesMatchingCssSelectors === false) {
-                continue;
+            // There's no real way to test "PHP Warning" output generated by the following XPath query unless PHPUnit converts it to an exception.
+            // Unfortunately, this would only apply to tests and not work for production executions, which can still flood logs/output unnecessarily.
+            // Instead, Emogrifier's error handler should always throw an exception and it must be caught here and only rethrown if in debug mode.
+            try {
+              // query the body for the xpath selector
+              $nodesMatchingCssSelectors = $xPath->query($this->translateCssToXpath($cssRule['selector']));
```

Comment:
I think we should have a comment here explaining that this will never be false as we already handle this with via the error handler.